zipline/utils/pandas_utils.py

```diff
@@ -222,3 +223,45 @@ def clear_dataframe_indexer_caches(df):
             delattr(df, attr)
         except AttributeError:
             pass
+
+
+def categorical_df_concat(df_list, inplace=False):
+    """
+    Prepare list of pandas DataFrames to be used as input to pd.concat.
+    Ensure any columns of type 'category' have the same categories across each
+    dataframe.
+
+    Parameters
+    ----------
+    df_list : list
+        List of dataframes with same columns.
+    inplace : bool
+        True if input list can be modified. Default is False.
+
+    Returns
+    -------
+    concatenated : df
+        Dataframe of concatenated list.
+    """
+
+    if not inplace:
+        df_list = deepcopy(df_list)
+
+    # Assert each dataframe has the same columns/dtypes
+    df = df_list[0]
+    if not all([(df.dtypes.equals(df_i.dtypes)) for df_i in df_list[1:]]):
+        raise ValueError("Input DataFrames must have the same columns/dtypes.")
+
+    categorical_columns = df.columns[df.dtypes == 'category']
+
+    for col in categorical_columns:
+        new_categories = sorted(
+            set().union(
+                *(frame[col].cat.categories for frame in df_list)
+            ) - {None}
```

Comment:
>  It doesn't matter regardless because an array entry missing a categorical code is assumed to be `None`.

To be clear, I don't think this is true; you'd get `NaN` back, rather than `None`, which has very different behavior.  In particular, `NaN` compares unequal with any other python object, including itself, which means you need to test for its presence explicitly.